accepted/2021/System.Diagnostics/Metrics-Design.md

```diff
@@ -0,0 +1,511 @@
+        /// </summary>
+        public bool Enabled => throw null;
+
+        /// <summary>
+        /// A property tells if the instrument is a regular instrument or an observable instrument.
+        /// </summary>
+        public virtual bool IsObservable => throw null;
+    }
+}
+```
+
+### Non-observable Instrument Base Class
+
+```csharp
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// Instrument<T> is the base class from which all non-observable instruments will inherit from.
+    /// Mainly It'll support the CLS compliant numerical types
+    /// </summary>
+    public abstract class Instrument<T> : Instrument where T : unmanaged
+    {
+        /// <summary>
+        /// Protected constructor to create the instrument with the common properties.
+        /// </summary>
+        protected Instrument(Meter meter, string name, string? description, string? unit) :
+                        base(meter, name, description, unit) { throw null; }
+
+        /// <summary>
+        /// Record measurement overloads allowing passing different numbers of tags.
+        /// </summary>
+
+        protected void RecordMeasurement(T val) { throw null; }
+
+        protected void RecordMeasurement(
+                            T val,
+                            KeyValuePair<string, object?> tag1) { throw null; }
+
+        protected void RecordMeasurement(
+                            T val,
+                            KeyValuePair<string, object?> tag1,
+                            KeyValuePair<string, object?> tag2) { throw null; }
+
+        protected void RecordMeasurement(
+                            T val,
+                            KeyValuePair<string, object?> tag1,
+                            KeyValuePair<string, object?> tag2,
+                            KeyValuePair<string, object?> tag3) { throw null; }
+
+        protected void RecordMeasurement(
+                            T val,
+                            ReadOnlySpan<KeyValuePair<string, object?>> tags) { throw null; }
+    }
+}
+```
+
+### Observable Instrument Base class
+
+```csharp
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// ObservableInstrument<T> is the base class from which all observable instruments will inherit from.
+    /// Mainly It'll support the CLS compliant numerical types
+    /// </summary>
+    public abstract class ObservableInstrument<T> : Instrument where T : unmanaged
+    {
+        /// <summary>
+        /// Protected constructor to create the instrument with the common properties.
+        /// </summary>
+        protected ObservableInstrument(
+                    Meter meter,
+                    string name,
+                    string? description,
+                    string? unit) : base(meter, name, description, unit) { throw null; }
+
+        /// <summary>
+        /// Observe will report the non-additive values of the measurements.
```

Comment:
```suggestion
        /// Observe() fetches the current measurements being tracked by this instrument.
```
(The text about non-additive appears arbitrary, there are additive observable instruments as well)